src/map/map.ts

```diff
@@ -1044,10 +1048,13 @@ export class PropertiesMap {
         }
 
         // switch all traces to 3D mode
-        this._restyle(({
-            'marker.symbol': symbols,
-            type: 'scatter3d',
-        } as unknown) as Data);
+        this._restyle(
+            ({
+                'marker.symbol': symbols,
+                type: 'scatter3d',
+            } as unknown) as Data,
+            [0, 1, 2]
```

Comment:
We don't want to only change the trace type and symbol for traces 0, 1, 2, we want to change it for all traces, including the ones used to display the symbol legend. I'm having the same issue in a separate branch, so I'll look for a fix.